Topic/kp rdkb 59901

dharshini555 opened the pull request.

dharshini555 committed:
RDKB-59901: KP reboot due to telemetry loss with rt timeout

Reason for change: Testing purpose
Test Procedure: There should not be crash and regression issues
Risks: Medium
Priority: P0
Signed-off-by: PriyaDharshini_Kathiravan <priyakathiravan05@gmail.com>

dharshini555 committed:
RDKB-59901: KP reboot due to telemetry loss with rt timeout

Reason for change: Testing purpose
Test Procedure: There should not be crash and regression issues
Risks: Medium
Priority: P0
Signed-off-by: PriyaDharshini_Kathiravan <priyakathiravan05@gmail.com>

dharshini555 committed:
RDKB-59901: KP reboot due to telemetry loss with rt timeout

Reason for change: Testing purpose
Test Procedure: There should not be crash and regression issues
Risks: Medium
Priority: P0
Signed-off-by: PriyaDharshini_Kathiravan <priyakathiravan05@gmail.com>

dharshini555 committed:
Update t2eventreceiver.c

dharshini555 committed:
Update rbusInterface.c

shibu-kv commented on source/ccspinterface/rbusInterface.c:
```diff
@@ -244,11 +246,30 @@ Vector* getRbusProfileParamValues(Vector *paramList)
         {
             T2Debug("Calling rbus_getExt for %s \n", paramNames[0]);
-            if(RBUS_ERROR_SUCCESS != rbus_getExt(t2bus_handle, 1, (const char**)paramNames, &paramValCount, &rbusPropertyValues))
+            int retries = 0;
```
We shouldn't do retries for timeouts. Timeouts can be for legit cases as well.

shibu-kv commented on source/ccspinterface/rbusInterface.c:
```diff
@@ -856,7 +877,7 @@ rbusError_t t2PropertyDataGetHandler(rbusHandle_t handle, rbusProperty_t propert
 
 
-void publishReportUploadStatus(char* status)
+void publishReportUploadStatus(char* status) //needs revisit
```
This is a tricky handling and would need some refactoring. For the time being keep a local copy of data in the function an release once done.

shibu-kv reviewed (changes requested):
In general please do address coverity defects.

dharshini555 committed:
RDKB-59901: KP reboot due to telemetry loss with rt timeout

Reason for change: Testing purpose
Test Procedure: There should not be crash and regression issues
Risks: Medium
Priority: P0
Signed-off-by: PriyaDharshini_Kathiravan <priyakathiravan05@gmail.com>

dharshini555 committed:
Merge branch 'topic/kp-RDKB-59901' of https://github.com/rdkcentral/telemetry into topic/kp-RDKB-59901

rdkcmf-jenkins commented on source/bulkdata/t2eventreceiver.c:
```diff
@@ -147,14 +181,22 @@
+                            T2Error("%s pthread_cond_signal for erCond failed with error code : %d\n", __FUNCTION__, ret);
+                        }
                     }
+                    pthread_mutex_unlock(&sTDMutex); //unlock for stopDispatchThread
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
ee0041243e88d7b2368d4c98bcc86d84
PRESENT
-->

# Coverity Issue - Double unlock
"pthread_mutex_unlock" unlocks "sTDMutex" while it is unlocked.

_Medium Impact, CWE-765_
_LOCK_

rdkcmf-jenkins commented on source/bulkdata/t2eventreceiver.c:
```diff
@@ -92,10 +105,16 @@
+                                }
+
                             }
+                            pthread_mutex_unlock(&sTDMutex); //unlock for stopDispatchThread
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
934866a6cdb9769979ca83a9ab00e0e4
PRESENT
-->

# Coverity Issue - Double unlock
"pthread_mutex_unlock" unlocks "sTDMutex" while it is unlocked.

_Medium Impact, CWE-765_
_LOCK_
